### ISSUE TITLE
API for adding resources to watches isn't intuitive

### ISSUE THREAD

**Issue by jasonwbarnett** [jfrog/terraform-provider-artifactory#82]
Currently the API for adding a repository (or any type of resource) is like so:

```terraform
resource "artifactory_xray_watch" "test" {
  name        = "watch-npm-local-repo"
  description = "apply a severity-based policy to the npm local repo"

  resources {
    type       = "repository"
    name       = artifactory_local_repository.npm-local.key
    bin_mgr_id = "example-com-artifactory-instance"
    repo_type  = "local"
  }
}
```

I strongly believe that the `resources` key should be singular (i.e. `resource`) since it is intended to be specified once per resource you intend to add to the xray watch. This would make this more consistent with the core terraform modules.

**Comment by skyzyx** [comment]
I imagine this is a breaking change, so it should be deferred to v3.0.0. But I think it's a good idea.

**Comment by chb0github** [comment]
Yes, this would be breaking. I have actually spoken @jasonwbarnett. Right now I have an open discussion ticket about moving the xray code out into it's own provider. And, at that point, I think it would be a fair discussion to change the HCL. 

I didn't come up with this HCL, and my editor gives me warnings about it. But I can't just break people. Please upvote and give your comments on the discussion to move xray to it's own provider
